src/main/java/uk/gov/homeoffice/digital/sas/timecard/validators/timeentry/TimeEntryValidator.java

```diff
@@ -20,8 +21,47 @@ public class TimeEntryValidator implements ConstraintValidator<TimeEntryConstrai
   @Override
   public boolean isValid(TimeEntry timeEntry, ConstraintValidatorContext context) {
 
-    EntityManager entityManager = BeanUtil.getBean(EntityManager.class);
-    Session session = entityManager.unwrap(Session.class);
+    if (timeEntry.getActualEndTime() != null
+        && timeEntry.getActualStartTime().after(timeEntry.getActualEndTime())) {
+      var message = ErrorMessage.END_TIME_BEFORE_START_TIME.toString();
+      var errorProperty = Property.END_TIME;
+
+      addConstraintViolationToContext(context, message, errorProperty, null);
+      return false;
+    }
+
+    var timeEntryClashes = getClashingTimeEntries(timeEntry);
+    if (!timeEntryClashes.isEmpty()) {
+      var clashingProperty = getClashingProperty(timeEntry, timeEntryClashes);
+      var payload = timeEntryClashes.stream().map(this::transformTimeEntry)
+          .collect(Collectors.toCollection(ArrayList::new));
+      var message = ErrorMessage.TIME_PERIOD_CLASH.toString();
+
+      addConstraintViolationToContext(context, message, clashingProperty, payload);
+      return false;
+    }
+    return true;
+  }
+
+  private static void addConstraintViolationToContext(ConstraintValidatorContext context,
```

Comment:
Was curious about the use of static here (and below method). Seen it on another PR too. Discussed with @taomoh and it seems like theres no harm in using it here but also there doesn't seem to be a need for it either.

I get that there will only be one instance of that method in the whole application run (which is fine) as it doesn't use any instance variables.
However it's just a **private** method called solely by a **public non-static** method. We can see that the other private methods in the class are not static as expected.

Reply:
I've removed it now. I used the IntelliJ extract method rather than writing these myself, it seems it defaults to static.